etc/ci/performance/runner.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/python3
+import argparse
+import itertools
+import json
+import os
+import subprocess
+
+
+def load_manifest(filename):
+    with open(filename, 'r') as f:
+        text = f.read()
+    return list(parse_manifest(text))
+
+
+def parse_manifest(text):
+    return filter(lambda x: x != "" and not x.startswith("#"),
+                  map(lambda x: x.strip(), text.splitlines()))
+
+
+def test_load(url, timeout):
+    ua_script_path = "{}/user-agent-js".format(os.getcwd())
+    test_cmd = ("timeout {timeout}s ./servo/servo '{url}'"
+                "--userscripts {ua} -x -o {png}").format(timeout=timeout,
+                                                         url=url,
+                                                         ua=ua_script_path,
+                                                         png="output.png")
+
+    print("Running test:")
+    print(test_cmd)
+    print("Timeout:{}".format(timeout))
+    log = ""
+    try:
+        log = subprocess.check_output(test_cmd, stderr=subprocess.STDOUT,
+                                      shell=True, timeout=timeout)
+    except subprocess.CalledProcessError as e:
+        print("Unexpected Fail:")
+        print(e)
+        print("You man want to re-run the test manually:\n{}".format(test_cmd))
+    except subprocess.TimeoutExpired:
+        print("Test timeout: {}".format(url))
+
+    return log
+
+
+def parse_log(log):
+    blocks = []
+    block = []
+    copy = False
+    for line_bytes in log.splitlines():
+        line = line_bytes.decode()
+
+        if line.strip() == ("[PERF] perf block start"):
+            copy = True
+        elif line.strip() == ("[PERF] perf block end"):
+            copy = False
+            blocks.append(block)
+            block = []
+        elif copy:
+            block.append(line)
+
+    def parse_block(block):
+        timing = {}
+        for line in block:
+            key = line.split(",")[1]
+            value = line.split(",")[2]
+
+            if key == "testcase":
+                timing[key] = value
+            else:
+                timing[key] = None if (value == "undefined") else int(value)
+        return timing
+
+    if len(blocks) == 0:
+        print("Didn't find any perf data in the log, test timeout?")
+    timings = map(parse_block, blocks)
+    return timings
+
+
+def filter_result_by_manifest(result_json, manifest):
+    print(manifest)
+    print(result_json)
+    return [tc for tc in result_json if tc['testcase'] in manifest]
+
+
+def median(lst):
```

Comment:
https://docs.python.org/3/library/statistics.html#statistics.median does this already, the downside is afaik it's not available for python 2, but since this is python 3, it should be fine